pkg/beholder/auth_test.go

```diff
@@ -2,21 +2,137 @@ package beholder
 	}
 
 	assert.Equal(t, expectedHeaders, BuildAuthHeaders(csaPrivKey))
 }
+
+func TestBuildAuthHeadersV2(t *testing.T) {
+	csaPrivKey, err := generateTestCSAPrivateKey()
+	require.NoError(t, err)
+	timestamp := time.Now().UnixMilli()
+
+	authHeaderMap := BuildAuthHeadersV2(csaPrivKey, &AuthHeaderConfig{
+		timestamp: timestamp,
+	})
+
+	authHeaderValue, ok := authHeaderMap[authHeaderKey]
+	require.True(t, ok, "auth header should be present")
+
+	parts := strings.Split(authHeaderValue, ":")
+	assert.Len(t, parts, 4, "auth header v2 should have 4 parts")
+	// Check the parts
+	version, pubKeyHex, timestampStr, signatureHex := parts[0], parts[1], parts[2], parts[3]
+	assert.Equal(t, authHeaderVersion2, version, "BuildAuthHeadersV2 should should have version 2")
+	assert.Equal(t, hex.EncodeToString(csaPrivKey.Public().(ed25519.PublicKey)), pubKeyHex)
+	assert.Equal(t, strconv.FormatInt(timestamp, 10), timestampStr)
+
+	// Decode the public key and signature
+	pubKeyBytes, err := hex.DecodeString(pubKeyHex)
+	require.NoError(t, err)
+	assert.Equal(t, csaPrivKey.Public().(ed25519.PublicKey), ed25519.PublicKey(pubKeyBytes))
+
+	// Parse the timestamp
+	timestampParsed, err := strconv.ParseInt(timestampStr, 10, 64)
+	require.NoError(t, err)
+	assert.Equal(t, timestamp, timestampParsed)
+	timestampBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(timestampBytes, uint64(timestampParsed))
+
+	// Reconstruct the message bytes
+	messageBytes := append(pubKeyBytes, timestampBytes...)
+
+	// Verify the signature
+	signatureBytes, err := hex.DecodeString(signatureHex)
+	require.NoError(t, err)
+	assert.True(t, ed25519.Verify(pubKeyBytes, messageBytes, signatureBytes))
+}
+
+func TestBuildAuthHeadersV2WithDefaults(t *testing.T) {
+	csaPrivKey, err := generateTestCSAPrivateKey()
+	require.NoError(t, err)
+
+	authHeaderMap := BuildAuthHeadersV2(csaPrivKey, nil)
+	authHeaderValue, ok := authHeaderMap[authHeaderKey]
+	require.True(t, ok, "auth header should be present")
+
+	parts := strings.Split(authHeaderValue, ":")
+	assert.Len(t, parts, 4, "auth header v2 should have 4 parts")
+	// Check the parts
+	version, pubKeyHex, timestampStr, signatureHex := parts[0], parts[1], parts[2], parts[3]
+	assert.Equal(t, "2", version, "using WithAuthHeaderV2 should should have version 2")
+	assert.Equal(t, hex.EncodeToString(csaPrivKey.Public().(ed25519.PublicKey)), pubKeyHex)
+
+	// Decode the public key and signature
+	pubKeyBytes, err := hex.DecodeString(pubKeyHex)
+	require.NoError(t, err)
+	assert.Equal(t, csaPrivKey.Public().(ed25519.PublicKey), ed25519.PublicKey(pubKeyBytes))
+
+	// Parse the timestamp
+	timestampParsed, err := strconv.ParseInt(timestampStr, 10, 64)
+	require.NoError(t, err)
+
+	// Verify the timestamp is within the last 5 seconds
+	// This verifies that default configuration is to use the current time
+	now := time.Now().UnixMilli()
+	assert.InDelta(t, now, timestampParsed, 5000, "timestamp should be within the last 5 seconds")
```

Comment:
nitpick: could tighten up the time range by recording `now` at the start of the test, sleeping a few milliseconds, and then comparing against that